### PR TITLE
[WIP] CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by weedySeaDragon**
Here's a first draft of CONTRIBUTING.md - the guidelines for contributing to the this project.

I copied this from AgileVentures/projectscope and made some minor changes for readability and reorganization (and so it's specifically for this project).
### I have left _TODO..._ questions/ comments in this.  They are:
- Do people have to join AV in order to contribute to this project?
- Can anyone that is a member of this project create a feature (user story) in PivotalTracker? 
- Can anyone that is a member of this project create an issue in GitHub?
-  How are issues in GitHub kept in sync with what's in PivotalTracker? 
- (under 'bug fix) Where should the example user story go? _(This is what happens when you blindly copy a file from another repo!)_  Should this story go in the body of the github issue? In the code? 
### I also created some diagrams to show both the process for joining the project and the GitHub workflow.
1.  I'm not sure if they're needed, but they're an option.  (I think they're useful.)
   -  I used [PlantUML](http://plantuml.com/) to generate them.  The control over formatting is limited, but it's free and having the source as plain text is handy. (It's rather like markdown for diagrams.)  I used a RubyMine plugin and did them right in my RubyMine IDE, btw.
2. _Most importantly_ **Are the processes (flow) correct?**  Since I'm new to AV, I'm coming up a learning curve on all of this.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  - [GitHub Workflow](#GitHub-Workflow)
+  - [Code Style](#Code-Style)      
+  
+  
+---
+
+
+## Defining Tasks with PivotalTracker and GitHub
+
+We use [GitHub Issues](https://github.com/AgileVentures/shf-project/issues) to manage our work on features, chores and bugfixes.
```
Comment (thesuss) [inline CONTRIBUTING.md:44]: I think we simply use Pivotal tracker for this?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+### Bug fixes
+
+Fixing a bug should **start with the creation of a test that replicates the bug,** so that any bugfix submission will include an appropriate test as well as the fix itself.
+
+A bugfix may include an acceptance test depending on where the bug occurred.
+
+
+
+_TODO_  _where should this user story go in the code?  as a cucumber acceptance test or as a documentation block?  in the github issue?_
```
Comment (thesuss) [inline CONTRIBUTING.md:64]: It is for features in cucumber. Yes.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+```
+
+Note that you should be re-syncing daily (even hourly at very active times) on your feature/bugfix branch to ensure that you are always building on top of very latest _develop_ code.
+
+ 
+#### Create a new branch for your work: [GitHub issue #]_[short description]
+ 
+When you create a branch to work on your feature or bug-fix, the name of the branch should *start with the GitHub issue #,* followed by an underscore, then a few words that describe the issue (i.e. from the title of the issue).  
+
+The format for branch names is [GitHub issue #]_[short description]
```
Comment (thesuss) [inline CONTRIBUTING.md:115]: We don't use GH issues, we could use PT story numbers, but I dont think that is necessary?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+Here is [more information on creating (submitting) pull requests](how_to_submit_a_pull_request_on_github.md).
+
+
+##### One change per PR
+
+Please ensure that each commit in your pull request makes a _single_ coherent change and that the overall pull request only includes commits related to the specific GitHub issue that the pull request is addressing.
+This helps the project managers understand the PRs and merge them more quickly.
+
+
+##### Every Pull Request (PR) should refer to a corresponding GitHub issue.
```
Comment (thesuss) [inline CONTRIBUTING.md:154]: Corresponding PIV feature/bug/chore

**Review by tochman — APPROVED** [review]
IMO we can merge this and deal with the outstanding TODO´s as separate chores.

**Pull request merged**